File: java/001/Main.java
/*
 * ProjectEuler 001
 * Java / Main.java
 * copyright (c) 2015 Susisu
 *
 * Java SE 1.8.0_25
 * `javac Main.java -d bin`
 */

public class Main {
    public static void main(String[] args) {
        int sum = 0;
        for (int n = 1; n < 1000; n++) {
            if (n % 3 == 0 || n % 5 == 0) {
                sum += n;
            }
        }
        System.out.println(sum);
    }
}

File: haskell/001/Main.hs
{-
    ProjectEuler 001
    Haskell / Main.hs
    copyright (c) 2015 Susisu

    GHC 7.8.3
    `ghc Main.hs -o bin/main`
-}

module Main where

main :: IO ()
main = print $ calc 1 999 0

calc :: Int -> Int -> Int -> Int
calc n max sum
    | n <= max =
        if n `mod` 3 == 0 || n `mod` 5 == 0 then
            calc (n + 1) max (sum + n)
        else
            calc (n + 1) max sum
    | otherwise = sum
